elf.go

```diff
@@ -215,8 +215,9 @@ func (ec *elfCode) loadMaps(mapSections map[int]*elf.Section) (map[string]*MapSp
 				return nil, errors.Errorf("map %v: can't read inner map index", name)
 			}
 
-			if rd.Len() != 0 {
-				return nil, errors.Errorf("map %v: unknown fields in definition", name)
+			var unknown uint32
+			if rd.Len() > 0 && binary.Read(rd, ec.ByteOrder, &unknown) != nil || unknown != 0 {
```

Comment:
Suggestion: use something like `bytes.Count(rd.Bytes(), []byte{0}) != rd.Len()`. Then we handle arbitrary zero tails.